app/src/utils/ocelloids.ts

```diff
@@ -0,0 +1,286 @@
+import { captureException } from '@sentry/nextjs'
+import {
+  CompletedTransfer,
+  OngoingTransferWithDirection,
+  StoredTransfer,
+  TxStatus,
+} from '@/models/transfer'
+import { OcelloidsAgentApi, OcelloidsClient, xcm } from '@sodazone/ocelloids-client'
+import { getExplorerLink, isParachainToParachain } from './transfer'
+import { NotificationSeverity, Notification } from '@/models/notification'
+import { Direction, resolveDirection } from '@/services/transfer'
+import { Polkadot } from '@/registry/mainnet/tokens'
+
+type ResultNotification = {
+  message: string
+  severity: NotificationSeverity
+  status: TxStatus
+}
+
+const OCELLOIDS_API_KEY = process.env.NEXT_PUBLIC_OC_API_KEY_READ_WRITE || ''
+
+// TMP Helper until Ocelloids supports DOT transfers between non system chains.
+export const isTransferringDotBetweenParachains = (
+  transfer: StoredTransfer | OngoingTransferWithDirection,
+): boolean => {
+  return transfer.token.id === Polkadot.DOT.id && isParachainToParachain(transfer)
+}
+
+// Helper to filter the subscribable transfers only
+export const getSubscribableTransfers = (transfers: StoredTransfer[]) => {
+  if (transfers.length === 0) return []
+  return transfers.filter(t => {
+    // Filters XCM transfers only
+    if (resolveDirection(t.sourceChain, t.destChain) === Direction.WithinPolkadot) {
+      // Exclude transfer if DOT is transfered between parachains
+      if (isTransferringDotBetweenParachains(t)) return false
+      return true
+    }
+    return false
+  })
+}
+
+export const initOcelloidsClient = () => {
+  if (!OCELLOIDS_API_KEY || !OCELLOIDS_API_KEY.length)
```

Comment:
```suggestion
  if (!OCELLOIDS_API_KEY)
```
an empty string is false